### PR TITLE
Add metrics for deposits and kobo

### PR DESCRIPTION
**Pull request by avdata99**
Related to #812 

This issue require to get information that is probably not available directly
Number of deposited datasets

image

Kobo packages and resources

image

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Next release - 202X-XX-XX
 
 New features:
-
+ - Definie metrics about deposited datasets [#951](https://github.com/okfn/ckanext-unhcr/pull/951)
```
Comment (amercader) [inline CHANGES.md:6]:
```suggestion
 - Show metrics about deposited datasets [#951](https://github.com/okfn/ckanext-unhcr/pull/951)
```

File: ckanext/unhcr/metrics/activity.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    results = []
+    for activity in activities.all():
+        if not activity.data.get(object_type):
+            continue
+        data = activity.data.get(object_type)
+        for flt in filters:
+            if data.get(flt) != filters[flt]:
+                break
+        else:
+            results.append(activity)
```
Comment (amercader) [inline ckanext/unhcr/metrics/activity.py:42]: This is going to be really slow, we need to do this query at the db level.
To filter by something in the data field you need to cast it as JSON, eg for object type:

```python
from sqlalchemy.types import JSON

activities = activities.filter(cast(model.Activity.data, JSON).op("->>")(object_type)!=None)
```

File: ckanext/unhcr/tests/pt/test_metrics_activity.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    get_curation_activities_by_year,
+    get_deposited_datasets_by_year,
+    list_activities,
+)
+from ckanext.unhcr.activity import create_curation_activity
+
+
+@pytest.mark.usefixtures('clean_postgis', 'clean_db', 'clean_index', 'unhcr_migrate')
+class TestMetricsActivity:
+    def setup(self):
```
Comment (amercader) [inline ckanext/unhcr/tests/pt/test_metrics_activity.py:16]: Don't use setup methods as they don't work well with pytest. Use fixtures like here:

https://github.com/ckan/ckan/wiki/CKAN-2.9-to-2.10-migration-tips#test-classes-setup-methods-not-firing-or-firing-in-wrong-order

File: ckanext/unhcr/metrics/activity.py (hunk 2 above)
Comment (avdata99) [reply ckanext/unhcr/metrics/activity.py]: Updated [here](https://github.com/okfn/ckanext-unhcr/commit/e166ec12490a1bddaafa4b99375b8195688b8394) @amercader. Also added the fiters for sub-items in the `Activity.data` dict

File: ckanext/unhcr/tests/pt/test_metrics_activity.py (hunk 3 above)
Comment (avdata99) [reply ckanext/unhcr/tests/pt/test_metrics_activity.py]: Updated tests [here](https://github.com/okfn/ckanext-unhcr/commit/b6864f61f4d29ad62c59213fbd386682d8bd0db3) and [here](https://github.com/okfn/ckanext-unhcr/commit/e4842bbe965d1936e0b41a18eda6928b518a5866) @amercader

**Review by amercader — APPROVED** [review]

**Pull request merged**